src/NSwag.CodeGeneration.CSharp/CSharpGeneratorBaseSettings.cs

```diff
@@ -49,6 +51,22 @@ protected CSharpGeneratorBaseSettings()
         [JsonIgnore]
         public override CodeGeneratorSettingsBase CodeGeneratorSettings => CSharpGeneratorSettings;
 
+        /// <summary>Gets or sets a value indicating whether to generate C# style naming (default: false).</summary>
+        public bool CSharpNamingConvention
+        {
+            get => _cSharpNamingConvention;
+            set
+            {
+                _cSharpNamingConvention = value;
+
+                // Install the C# style name generators if desired
+                if (!_cSharpNamingConvention) return;
+                CSharpGeneratorSettings.PropertyNameGenerator = new CSharpPropertyNameGenerator();
+                CSharpGeneratorSettings.TypeNameGenerator = new CSharpTypeNameGenerator();
+                CSharpGeneratorSettings.EnumNameGenerator = new CSharpEnumNameGenerator();
```

Comment:
I think these three implementations should not be added here but we should change the defaults in NJsonSchema. This just adds another layer of complexity and if you need to do that you already have the CSharpGeneratorSettings class as the extension point. Please create a PR in NJsonSchema with your improvements.